example-site/components/TableErrorValidation/TableWithValidation.tsx

```diff
@@ -0,0 +1,373 @@
+		size: '',
+		label: 'Upload',
+		buttonId: 'vehicle-build-and-layout-plans-upload-button',
+		error: false,
+	},
+	{
+		documentType: 'Food Safety Supervisor Certificate for Alex',
+		file: 'filename.pdf',
+		size: '88kb',
+		label: 'Upload',
+		buttonId: 'food-safety-supervisor-certificate-for-alex-upload-button',
+		error: false,
+	},
+	{
+		documentType: 'Suggested menu or list of foods being sold',
+		file: 'filename.pdf',
+		size: '88kb',
+		label: 'Upload',
+		buttonId: 'suggested-menu-or-list-of-foods-being-sold-upload-button',
+		error: false,
+	},
+];
+export type AssessmentFile = (typeof defaultAssessmentFiles)[number];
+
+const locationPageError = {
+	href: `#${locationTextInputId}`,
+	message: 'Location of assessment',
+	name: 'location' as const,
+};
+
+const datePageError = {
+	href: `#${dateOfAssessmentDatePickerId}`,
+	message: 'Date of assessment',
+	name: 'date' as const,
+};
+
+const filesPageError = {
+	href: `#section-alert`,
+	message: 'Assessment files',
+	name: 'files' as const,
+};
+
+type PageError =
+	| typeof locationPageError
+	| typeof datePageError
+	| typeof filesPageError;
+
+// const getPageErrors = ({name}: {name: PageErrors[number]['name']}) => {
+
+// }
+
+const sampleTableError = {
+	href: '#vehicle-build-and-layout-plans-upload-button',
+	message: 'Upload files',
+	name: 'files',
+};
+type SampleTableError = typeof sampleTableError;
+
+export const TableWithValidation = ({ tableRef }: MultiTablePageProps) => {
+	const tableSectionAlertRef = useRef<HTMLDivElement>(null);
+
+	// Input data
+	const [dateValue, setDateValue] = useState<Date>();
+	const [locationValue, setLocationValue] = useState('');
+
+	// Table data
+	const [assessmentFiles, setAssessmentFiles] = useState(
+		defaultAssessmentFiles
+	);
+	const [currentAssessmentFile, setCurrentAssessmentFile] =
+		useState<AssessmentFile>();
+
+	// Errors
+	const [pageErrors, setPageErrors] = useState<PageError[]>([]);
+	const [tableErrors, setTableErrors] = useState<SampleTableError[]>([]);
+
+	// Drawer state
+	const [isDrawerOpen, openDrawer, closeDrawer] = useTernaryState(false);
+
+	// Action handlers (includes some mocking)
+	const mockUploadFile = () => {
+		const newAssessmentFiles = assessmentFiles.map((prevAssessmentFile) =>
+			currentAssessmentFile?.documentType === prevAssessmentFile.documentType
+				? {
+						...prevAssessmentFile,
+						file: 'filename.pdf',
+						size: '88kb',
+						error: false,
+				  }
+				: prevAssessmentFile
+		);
+		setAssessmentFiles(newAssessmentFiles);
+
+		setTableErrors(
+			newAssessmentFiles.reduce<SampleTableError[]>(
+				(acc, assessmentFile) =>
+					assessmentFile.file
+						? acc
+						: [
+								...acc,
+								{
+									href: `#${assessmentFile.buttonId}`,
+									message: `Upload a file for ${assessmentFile.documentType}`,
+									name: 'files',
+								},
+						  ],
+				[]
+			)
+		);
+		closeDrawer();
+	};
+
+	const mockUploadError = () => {
```

Comment:
What are we demonstrating with the button in the drawer here? What scenario would create an error like that in the drawer? Should this be moved the Save buttons?

Reply:
Interesting. I see what you're saying. Seems like my understanding of the process is different. I was thinking:

1. Press "Upload"
_Drawer opens_
2. Select a file (drag or browse)
_On select, the drawer closes and the file begins uploading_
3. Errors are communicated on the main page.

I figured we would actually upload files at each interaction rather than wait for a user to select, say, 10x 5mb reports and wait for each one to upload after clicking "Save and continue". We label the buttons "Upload" so that pointed me in this direction.

Anyway, let me know your thoughts, maybe we can jam on this as a team?